File: src/main/java/com/ggardet/doctor/internal/entity/Doctor.java
package com.ggardet.doctor.internal.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

import java.util.UUID;

@Entity
@Data
@Table(name = "doctor", schema = "doctors")
public class Doctor {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String firstname;

    @Column(nullable = false)
    private String lastname;

    @Column(nullable = false)
    private String specialty;

    @Column(name = "license_number")
    private String licenseNumber;

    @Column(name = "phone_number")
    private String phoneNumber;

    @Column
    private String email;

    @Column(name = "hospital_id")
    private UUID hospitalId;

    @Column(name = "years_experience")
    private Integer yearsExperience;
}

File: src/main/kotlin/com/ggardet/doctor/internal/entity/Doctor.kt
package com.ggardet.doctor.internal.entity

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "doctor", schema = "doctors")
class Doctor {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: UUID? = null

    @Column(nullable = false)
    var firstname: String = ""

    @Column(nullable = false)
    var lastname: String = ""

    @Column(nullable = false)
    var specialty: String = ""

    @Column(name = "license_number")
    var licenseNumber: String? = null

    @Column(name = "phone_number")
    var phoneNumber: String? = null

    @Column
    var email: String? = null

    @Column(name = "hospital_id")
    var hospitalId: UUID? = null

    @Column(name = "years_experience")
    var yearsExperience: Int? = null

    // Equals and hashCode based on id
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as Doctor
        return id == other.id
    }

    override fun hashCode(): Int = id?.hashCode() ?: 0

    override fun toString(): String {
        return "Doctor(id=$id, firstname='$firstname', lastname='$lastname', specialty='$specialty')"
    }
}